### ISSUE TITLE
How to use and upload the customized my own pre-trained model for feature extraction? (customized)

### ISSUE THREAD

**Issue by kaen2891** [s3prl/s3prl#215]
How to use and upload the customized pre-trained model for feature extraction?

Hello, I trained my own dataset for pre-training with Mockingjay and TERA model as described in https://github.com/s3prl/s3prl/tree/master/s3prl/pretrain

And now, the model training is finished, and I got the ckpt files at `./result/pretrain/mockingjay/states-200000.ckpt`

Actually, what I want to do is upload my pre-trained model with ckpt files and extract the feature representation from the pre-trained model.  Such as below: 

```
import s3prl.hub as hub

model_0 = getattr(hub, 'fbank')()  # use classic FBANK
model_1 = getattr(hub, 'modified_cpc')()  # build the CPC model with pre-trained weights
model_2 = getattr(hub, 'tera')()  # build the TERA model with pre-trained weights
model_3 = getattr(hub, 'wav2vec2')()  # build the Wav2Vec 2.0 model with pre-trained weights

device = 'cuda'  # or cpu
model_3 = model_3.to(device)
wavs = [torch.randn(160000, dtype=torch.float).to(device) for _ in range(16)]
with torch.no_grad():
    reps = model_3(wavs)["hidden_states"]
```
It means that not using for downstream tasks in s3rpl.

To this end, I refer the `https://github.com/s3prl/s3prl/blob/master/s3prl/upstream/example/README.md` and typed the command as:

``` 
python3 run_downstream.py -m train -n moc_korean \
    -u customized_upstream \
    -k /NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/states-200000.ckpt \
    -g /NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/config_model.yaml \
    -d example
```

After this procedure, I think the downstream tasks are doing with train/dev/test as below:

![image](https://user-images.githubusercontent.com/46586785/136170700-c506f617-ec3c-408b-8ad5-2a9cdc359adc.png)

After this procedure, I think the downstream tasks are doing with train/dev/test as below:

As a result, if I want to upload my pre-trained model to the hub for use as the feature extraction, what should I do? 

Or is there any way to use the pre-trained model directly for feature extraction?

Can you help with this for me?

**Comment by kaen2891** [comment]
Now, I tried one more as below:

```
my_ckpt = '/NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/states-200000.ckpt'
my_config = '/NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/config_model.yaml'

import s3prl.hub as hub

model = getattr(hub, 'mockingjay_local')(my_ckpt, my_config)
```

Then I got an error like this:

![image](https://user-images.githubusercontent.com/46586785/136175948-5211fd5c-1519-4bf0-844d-91542ef8e010.png)

How can I fix this error? If this problem does not happen, I think I can do what I want.

Thanks a lot.

**Comment by leo19941227** [comment]
Hi! Thanks for the report!

I will look into this ASAP.
However, I happen to have an urgent mission this week. Will try to find some time for this.
Thanks again!

**Comment by andi611** [comment]
> ```
> my_ckpt = '/NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/states-200000.ckpt'
> my_config = '/NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/config_model.yaml'
> 
> import s3prl.hub as hub
> 
> model = getattr(hub, 'mockingjay_local')(my_ckpt, my_config)
> ```

I assume your `my_config` is a file similar to this one here: [config_model.yaml](https://github.com/s3prl/s3prl/blob/854aff0b3062fc2cff531401923b8745f64701e7/s3prl/pretrain/mockingjay/config_model.yaml)
If so, then you should not pass in the `my_config`, because it is already stored in `states-200000.ckpt`.
You can change to code to the following and it will work:
```
import s3prl.hub as hub
my_ckpt = '/NasData/s3prl/s3prl/result/pretrain/mockingjay_korean/states-200000.ckpt'
model = getattr(hub, 'mockingjay_local')(my_ckpt)
```

Note that the [`model_config` argument](https://github.com/s3prl/s3prl/blob/854aff0b3062fc2cff531401923b8745f64701e7/s3prl/upstream/mockingjay/expert.py#L26) should be a `.yaml` like this: [options.yaml](https://github.com/s3prl/s3prl/blob/854aff0b3062fc2cff531401923b8745f64701e7/s3prl/upstream/mockingjay/options.yaml)

Sorry for not making it clear.
Hope this helps.

Best,
Andy

**Comment by leo19941227** [comment]
This issue should be fixed by #313.
If not, feel free to re-open it, thanks!
